### PR TITLE
Create How to create a MetaMask Plugin

### PR DESCRIPTION
**Pull request by sazeleo**
This is the Outline for the tutorial. @ChrisChinchilla 
For the POC Demo, I am thinking of file sharing plugin. Please provide your view!

### CODE REVIEW

File: MetaMask Plugin/How to create a MetaMask Plugin [hunk 1]
```diff
@@ -0,0 +1,26 @@
+Introduction
```
Comment (ChrisChinchilla) [inline MetaMask Plugin/How to create a MetaMask Plugin]:
```suggestion
## Introduction
```

File: MetaMask Plugin/How to create a MetaMask Plugin [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+1. Metamask plugin what’s new? 
+2. How this will improve or advantages over the extension.
```
Comment (ChrisChinchilla) [inline MetaMask Plugin/How to create a MetaMask Plugin]: The guidelines say to not use unnecessary future tense, which in this case means:

```suggestion
2. How this improves the extension.
```

Metamask plugins already exist, so are not in the future, they are already improving Metamask

> grammar lesson over ;)

File: MetaMask Plugin/How to create a MetaMask Plugin [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+
+3.1 BreakDown of new Plugin APIs of Metamask. (Short Video/ Writeup)
```
Comment (ChrisChinchilla) [inline MetaMask Plugin/How to create a MetaMask Plugin]:
```suggestion
3.1 Break down of new the Plugin APIs of Metamask. (Short Video/ Writeup)
```

Any more detail here?

File: MetaMask Plugin/How to create a MetaMask Plugin [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+3.2 Explain Approach, and suggest concepts.
```
Comment (ChrisChinchilla) [inline MetaMask Plugin/How to create a MetaMask Plugin]: What do you mean here?

File: MetaMask Plugin/How to create a MetaMask Plugin [hunk 5]
```diff
@@ -0,0 +1,26 @@
+Introduction
+
+1. Metamask plugin what’s new?
```
Comment (ChrisChinchilla) [inline MetaMask Plugin/How to create a MetaMask Plugin]:
```suggestion
1. What's new with the Metamask plugin
```

But this is specifically about the APIs yes? We don't need to cover everything that's new with Metamask unless it relates to the plugin APIs.

File: MetaMask Plugin/How to create a MetaMask Plugin [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+1. Metamask plugin what’s new? 
+2. How this will improve or advantages over the extension.
+
+
+3.1 BreakDown of new Plugin APIs of Metamask. (Short Video/ Writeup)
+3.2 Explain Approach, and suggest concepts. 
+
+
+4. How to Switch from Extension to Plugin.
```
Comment (ChrisChinchilla) [inline MetaMask Plugin/How to create a MetaMask Plugin]:
```suggestion
4. How to switch from extension to plugin. 
```

I'm not sure what you mean here, we're not switching from anything, it's about creating a plugin for Metamask.

File: MetaMask Plugin/How to create a MetaMask Plugin [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+2. How this will improve or advantages over the extension.
+
+
+3.1 BreakDown of new Plugin APIs of Metamask. (Short Video/ Writeup)
+3.2 Explain Approach, and suggest concepts. 
+
+
+4. How to Switch from Extension to Plugin. 
+
+POC
```
Comment (ChrisChinchilla) [inline MetaMask Plugin/How to create a MetaMask Plugin]:
```suggestion
## POC
```

File: MetaMask Plugin/How to create a MetaMask Plugin [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+5. POC Demo (Walkthrough)
+5.1. Abstract level code description. 
+5.2. Plugin APIs involved.
+5.3. Implementation
+
+
+Ancillary Material 
+- Metamask Plugin Team
```
Comment (ChrisChinchilla) [inline MetaMask Plugin/How to create a MetaMask Plugin]: Not needed, this is a technical post, though you could mention the governance they discuss in the announcement post

**Comment by ChrisChinchilla** [comment]
Thanks @thesachinmittal added some feedback and edits 👍

File: MetaMask Plugin/How to create a MetaMask Plugin (hunk 3 above)
Comment (sazeleo) [reply MetaMask Plugin/How to create a MetaMask Plugin]: https://workflowy.com/s/plugin-concepts-coll/BmH5IUfRIHa6EmHJ

I will refer to this. For different plugin related concepts, and will give a breakdown of the plugin, and its use case and how a developer can hack on it. Just the high-level approach.

File: MetaMask Plugin/How to create a MetaMask Plugin (hunk 6 above)
Comment (sazeleo) [reply MetaMask Plugin/How to create a MetaMask Plugin]: Apparently, the Community will build the plugins to make it more user-friendly. Hence switching from extension to plugin.

File: MetaMask Plugin/How to create a MetaMask Plugin (hunk 8 above)
Comment (sazeleo) [reply MetaMask Plugin/How to create a MetaMask Plugin]: Okay, Got it!

**Comment by sazeleo** [comment]
Thanks for the edits! Please give me an ETA, and a communication channel to manage the process.

**Comment by ChrisChinchilla** [comment]
@thesachinmittal An ETA for what? And we're managing the process on GitHub, where we're already speaking and working

**Review by ChrisChinchilla — CHANGES_REQUESTED** [review]
If you're happy with my suggestions, can you approve/commit them and then request a new review please?

**Comment by sazeleo** [comment]
> If you're happy with my suggestions, can you approve/commit them and then request a new review please?

Made some changes according to your suggestions.

File: MetaMask Plugin/How to create a MetaMask Plugin (hunk 6 above)
Comment (ChrisChinchilla) [reply MetaMask Plugin/How to create a MetaMask Plugin]: Sorry, I still don't understand what you mean @thesachinmittal 

https://medium.com/metamask/introducing-the-next-evolution-of-the-web3-wallet-4abdf801a4ee

Metamask is building a plugin architecture that sits on top of a browser extension.

**Review by ChrisChinchilla — APPROVED** [review]
OK, I'm going to approve this so you can get started, but the outline is lacking in detail, so please checkin with your full article before you finish it so I can check it's heading in the right direction.

**Pull request merged**